pkg/storage/mto_solution.go

```diff
@@ -89,6 +89,28 @@ func MTOSolutionCreate(
 	return returned, nil
 }
 
+// MTOSolutionCreateAllowConflicts creates a new MTOSolution in the database, but
+// in the case of a conflict, instead just returns the conflicting row (and doesn't return an error)
+func MTOSolutionCreateAllowConflicts(
```

Comment:
see other comment, I think we might want to consider bulk addition instead of one at a time. I'm good with leaving a todo